packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -117,13 +118,12 @@ export const usePostModalNavigation = (
   const getPostPosition = () => {
     const isPost = (item: FeedItem) => item.type === 'post';
     const firstPost = items.findIndex(isPost);
+    const isLast = items.length - 1 === openedPostIndex;
     if (firstPost === openedPostIndex)
       return items.length - 1 === openedPostIndex
         ? PostPosition.Only
         : PostPosition.First;
-    // We do not set the Last position because the local last might not be the actual last.
-    // The setting can be updated once we update the API to return the information if the page is last or not.
-    return PostPosition.Middle;
+    return !canFetchMore && isLast ? PostPosition.Last : PostPosition.Middle;
```

Comment:
Seems like we already had that information.

Reply:
I think we should add the `isFetchingNextPage` here
```suggestion
    return isFetchingNextPage || (!canFetchMore && isLast) ? PostPosition.Last : PostPosition.Middle;
```

Reply:
added